lib/node/failure_detector.ml

```diff
@@ -168,3 +170,37 @@ let failure_detection node =
           Lwt_unix.sleep @@ Float.of_int t.config.protocol_period;
         ] in
     Lwt.return ()
+
+module PeerSet = Set.Make (struct
+  let compare = Stdlib.compare
+  type t = Peer.t
+end)
+
+let suspicious_detection node =
+  let open Peer in
+  let t = !node.failure_detector in
+  let suspicious_peers =
+    List.filter
+      (fun p -> p.status = Peer.Suspicious)
+      (Base.Hashtbl.data !node.peers) in
+  match List.length suspicious_peers with
+  | 0 -> Lwt.return ()
+  | _ ->
+    let _ = Lwt_unix.sleep @@ Float.of_int t.config.suspicion_time in
```

Comment:
I have no clue if this does anything if we never pass it to bind

Reply:
Bound it.